integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/async/websocket/WsServerFrameHandler.java

```diff
@@ -30,12 +31,37 @@
     private static final Logger log = LoggerFactory.getLogger(WsServerFrameHandler.class);
 
     @Override
-    public void channelRead(ChannelHandlerContext ctx, Object msg) {
-        if (msg instanceof WebSocketFrame) {
-            if (msg instanceof TextWebSocketFrame) {
-                log.info("TextWebSocketFrame received. Message: {}", ((TextWebSocketFrame) msg).text());
+    public void channelRead(ChannelHandlerContext ctx, Object frame) {
+
+        if (frame instanceof WebSocketFrame) {
+            if (frame instanceof TextWebSocketFrame) {
+                String msg = ((TextWebSocketFrame) frame).text();
+                if (msg.length() > 50) {
+                    log.info("TextWebSocketFrame with a large payload received.");
+                    ctx.channel().writeAndFlush(
+                            new TextWebSocketFrame("Message received: " + msg));
```

Comment:
Return smaller payloads to check if client sent payloads were throttled

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2736/commits/7048d5c96997f9537de523dba580cdd561454894